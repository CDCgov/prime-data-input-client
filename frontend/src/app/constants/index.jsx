export const COVID_RESULTS = {
  POSITIVE: "POSITIVE",
  NEGATIVE: "NEGATIVE",
  INCONCLUSIVE: "UNDETERMINED",
};

export const TEST_RESULT_VALUES = {
  0: COVID_RESULTS.NEGATIVE,
  1: COVID_RESULTS.POSITIVE,
  2: COVID_RESULTS.INCONCLUSIVE,
};
