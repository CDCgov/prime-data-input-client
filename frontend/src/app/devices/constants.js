export const DEVICES = {
  deviceId1: {
    displayName: "Abbott Id Now",
  },
  deviceId2: {
    displayName: "Abbott Id Later",
  },
};
