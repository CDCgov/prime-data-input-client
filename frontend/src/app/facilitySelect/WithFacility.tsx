/* eslint-disable graphql/template-strings */
import { useReactiveVar } from "@apollo/client";
import { useCallback, useMemo, FC } from "react";
import { useHistory } from "react-router-dom";

import {
  appConfig,
  facilities,
} from "../../storage/store";
import { useFacilities } from "../../hooks/useFacilities";
import { getFacilityIdFromUrl } from "../utils/url";

import FacilityPopup from "./FacilityPopup";
import FacilitySelect from "./FacilitySelect";

const Loading: FC<{}> = () => <p>Loading facility information...</p>;

interface Props {}

const WithFacility: FC<Props> = ({ children }) => {
  const history = useHistory();
  const { setCurrentFacility } = useFacilities(facilities);
  const { list, current } = useReactiveVar<FacilitiesState>(facilities);
  const { dataLoaded } = useReactiveVar<AppConfigState>(appConfig);

  const facilityFromUrl = useMemo(
    () => list.find((f) => f.id === getFacilityIdFromUrl()),
    [list]
  );

  const setFacilityProp = useCallback(
    (facilityId: string) => {
      history.push({ search: `?facility=${facilityId}` });
    },
    [history]
  );

  const setActiveFacility = useCallback(
    (facility: Facility) => {
      setCurrentFacility(facility);
      setFacilityProp(facility.id);
    },
    [setFacilityProp, setCurrentFacility]
  );

  if (!dataLoaded) {
    return <Loading />;
  }

  if (list.length === 0) {
    return (
      <FacilityPopup>
        <p>You do not have access to any facilities at this time.</p>
        <p>
          Ask an administrator to assign you access, then try logging in again.
        </p>
      </FacilityPopup>
    );
  }

  if (list.length === 1 && (!facilityFromUrl?.id || !current?.id)) {
    setActiveFacility(list[0]);
    return <Loading />;
  }

  if (facilityFromUrl?.id && !current?.id) {
    setActiveFacility(facilityFromUrl);
    return <Loading />;
  }

  if (facilityFromUrl?.id && current?.id) {
    return <>{children}</>;
  }

  return (
    <FacilitySelect facilities={list} setActiveFacility={setActiveFacility} />
  );
};

export default WithFacility;
