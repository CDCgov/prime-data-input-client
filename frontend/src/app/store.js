import thunk from "redux-thunk";
import storage from "redux-persist/lib/storage"; // defaults to localStorage for web
// import hardSet from "redux-persist/lib/stateReconciler/hardSet";
import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import { persistStore, persistReducer } from "redux-persist";

import patientReducer from "./patients/state/patientReducers";
import testResultReducer from "./testResults/state/testResultReducer";
import testQueueReducer from "./testQueue/state/testQueueReducer";
import { initialState } from "./fakeData/initialState";

const rootReducer = combineReducers({
  patients: patientReducer,
  testResults: testResultReducer,
  testQueue: testQueueReducer,
});

const persistConfig = {
  key: "reduxStoreRoot", // key used in localstorage to reference the persisted redux store
  storage, // the storage to use
  // stateReconciler: hardSet,
  // whitelist: ["patients"],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const configureStore = () => {
  return createStore(
    persistedReducer,
    initialState,
    compose(
      applyMiddleware(thunk),
      (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()) ||
        compose
    )
  );
};

export const store = configureStore();
export const persistor = persistStore(store);
