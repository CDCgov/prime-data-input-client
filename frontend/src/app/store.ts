import { createStore } from "redux";

const SET_INITIAL_STATE = "SET_INITIAL_STATE";
const UPDATE_ORGANIZATION = "UPDATE_ORGANIZATION";

// this should be the default value for a brand new org
// TODO: get the fields from a schema or something; hard-coded fields are hard to maintain
const initialState = {
  organization: {
    name: "",
  },
  facilities: [],
  facility: {
    id: "",
    name: "",
  },
  user: {
    id: "",
    firstName: "",
    middleName: "",
    lastName: "",
    suffix: "",
  },
};

const reducers = (state = initialState, action: any) => {
  switch (action.type) {
    case SET_INITIAL_STATE:
      return {
        ...state,
        ...action.payload,
      };
    case UPDATE_ORGANIZATION:
      return {
        ...state,
        organization: {
          ...state.organization,
          ...action.payload,
        }
      };
    default:
      return state;
  }
};

export const setInitialState = (initialState: any) => {
  return {
    type: SET_INITIAL_STATE,
    payload: initialState,
  };
};

export const updateOrganization = (organization: any) => {
  return {
    type: UPDATE_ORGANIZATION,
    payload: organization,
  };
};

const configureStore = () => {
  return createStore(reducers, initialState);
};

export const store = configureStore();
