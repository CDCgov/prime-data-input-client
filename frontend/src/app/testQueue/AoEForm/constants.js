// BEGIN things that should be service calls
const globalSymptomDefinitions = [
  { value: "426000000", label: "Fever over 100.4F" },
  { value: "103001002", label: "Feeling feverish" },
  { value: "43724002", label: "Chills" },
  { value: "49727002", label: "Cough" },
  { value: "267036007", label: "Shortness of breath" },
  { value: "230145002", label: "Difficulty breathing" },
  { value: "84229001", label: "Fatigue" },
  { value: "68962001", label: "Muscle or body aches" },
  { value: "25064002", label: "Headache" },
  { value: "36955009", label: "New loss of taste" },
  { value: "44169009", label: "New loss of smell" },
  { value: "162397003", label: "Sore throat" },
  { value: "68235000", label: "Nasal congestion" },
  { value: "64531003", label: "Runny nose" },
  { value: "422587007", label: "Nausea" },
  { value: "422400008", label: "Vomiting" },
  { value: "62315008", label: "Diarrhea" },
];

export const getSymptomList = () => globalSymptomDefinitions;
export const getTestTypes = () => [
  { label: "Molecular", value: "1" },
  { label: "Antigen", value: "2" },
  { label: "Antibody/Serology", value: "3" },
  { label: "Unknown", value: "4" },
];
export const getPregnancyResponses = () => [
  { label: "Yes", value: "77386006" },
  { label: "No", value: "60001007" },
  { label: "Would not state", value: "261665006" },
];
