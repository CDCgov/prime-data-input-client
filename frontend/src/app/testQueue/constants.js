export const QUEUE_NOTIFICATION_TYPES = {
  ADDED_TO_QUEUE__SUCCESS: 1,
  SUBMITTED_RESULT__SUCCESS: 2,
};
