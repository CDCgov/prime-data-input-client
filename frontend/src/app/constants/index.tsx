import { TestResult } from "../testQueue/QueueItem";

export const COVID_RESULTS: { [key: string]: TestResult } = {
  POSITIVE: "POSITIVE",
  NEGATIVE: "NEGATIVE",
  INCONCLUSIVE: "UNDETERMINED",
};

export const TEST_RESULT_VALUES = {
  0: COVID_RESULTS.NEGATIVE,
  1: COVID_RESULTS.POSITIVE,
  2: COVID_RESULTS.INCONCLUSIVE,
};

export const TEST_RESULT_DESCRIPTIONS = {
  NEGATIVE: "Negative",
  POSITIVE: "Positive",
  UNDETERMINED: "Inconclusive",
};
