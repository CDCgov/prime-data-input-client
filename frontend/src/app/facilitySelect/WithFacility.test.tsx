import React from "react";
import renderer from "react-test-renderer";

import { facilitySample } from "../../config/constants";
import { appConfig, facilities } from "../../storage/store";

import WithFacility from "./WithFacility";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe("WithFacility", () => {
  let component: any;
  beforeEach(() => {
    mockHistoryPush.mockClear();
    renderer.act(() => {});
  });

  describe("With zero facilities", () => {
    beforeEach(() => {
      renderer.act(() => {
        appConfig({
          ...appConfig(),
          dataLoaded: true,
          organization: { name: "Organization Name" },
          user: {
            ...appConfig().user,
            firstName: "Kim",
            lastName: "Mendoza",
          },
        });
        component = renderer.create(<WithFacility>App</WithFacility>);
      });
    });

    it("should notify user to contact an admin", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe("With one facility", () => {
    beforeEach(() => {
      renderer.act(() => {
        appConfig({
          ...appConfig(),
          dataLoaded: false,
          organization: { name: "Organization Name" },
          user: {
            ...appConfig().user,
            firstName: "Kim",
            lastName: "Mendoza",
          },
        });
        facilities({
          ...facilities(),
          availableFacilities: [
            { ...facilitySample, id: "1", name: "Facility 1" },
          ],
        });
        component = renderer.create(<WithFacility>App</WithFacility>);
      });
    });

    it("should render with a value", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
    it("should select the first facility", () => {
      expect(facilities().selectedFacility?.id).toEqual("1");
    });
    it("should push history", () => {
      expect(mockHistoryPush).toHaveBeenCalled();
    });
    it("should set the facility id search param", () => {
      expect(mockHistoryPush).toHaveBeenCalledWith({ search: "?facility=1" });
    });
  });

  describe("With two facilities", () => {
    beforeEach(() => {
      renderer.act(() => {
        appConfig({
          ...appConfig(),
          dataLoaded: true,
          organization: { name: "Organization Name" },
          user: {
            ...appConfig().user,
            firstName: "Kim",
            lastName: "Mendoza",
          },
        });
        facilities({
          ...facilities(),
          selectedFacility: { id: "", name: "", ...facilitySample },
          availableFacilities: [
            { ...facilitySample, id: "1", name: "Facility 1" },
            { ...facilitySample, id: "2", name: "Facility 2" },
          ],
        });

        component = renderer.create(<WithFacility>App</WithFacility>);
      });
    });

    it("should render with a value", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });

    describe("On facility select", () => {
      beforeEach(() => {
        renderer.act(() => {
          component.root.findAllByType("button")[0].props.onClick();
        });
      });

      it("should select the first facility", () => {
        expect(facilities().selectedFacility?.id).toEqual("1");
      });
      it("should push history once", () => {
        expect(mockHistoryPush).toHaveBeenCalledTimes(1);
      });
      it("should set the facility id search param", () => {
        expect(mockHistoryPush).toHaveBeenCalledWith({ search: "?facility=1" });
      });
    });
  });
});
