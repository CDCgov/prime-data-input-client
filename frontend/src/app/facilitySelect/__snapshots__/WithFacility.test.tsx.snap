// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithFacility With one facility should render with a value 1`] = `
<p>
  Loading facility information...
</p>
`;

exports[`WithFacility With two facilities should render with a value 1`] = `
<main
  className="prime-home"
  id="facility-select"
>
  <div
    className="grid-container"
  >
    <div
      className="grid-row position-relative"
    >
      <div
        className="prime-container usa-card__container"
      >
        <div
          className="usa-card__header"
        >
          <div>
            <h2>
              SimpleReport
            </h2>
            <div
              className="organization-name"
            >
              Organization Name
            </div>
          </div>
        </div>
        <div
          className="usa-card__body"
        >
          <p
            className="welcome"
          >
            Welcome, 
            Kim Mendoza
          </p>
          <p
            className="select-text"
          >
            Please select which facility you are working at today
          </p>
          <button
            className="usa-button usa-button--outline"
            onClick={[Function]}
            type="button"
          >
            Facility 1
          </button>
          <button
            className="usa-button usa-button--outline"
            onClick={[Function]}
            type="button"
          >
            Facility 2
          </button>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`WithFacility With zero facilities should notify user to contact an admin 1`] = `
<main
  className="prime-home"
  id="facility-select"
>
  <div
    className="grid-container"
  >
    <div
      className="grid-row position-relative"
    >
      <div
        className="prime-container usa-card__container"
      >
        <div
          className="usa-card__header"
        >
          <div>
            <h2>
              SimpleReport
            </h2>
            <div
              className="organization-name"
            >
              Organization Name
            </div>
          </div>
        </div>
        <div
          className="usa-card__body"
        >
          <p
            className="welcome"
          >
            Welcome, 
            Kim Mendoza
          </p>
          <p>
            You do not have access to any facilities at this time.
          </p>
          <p>
            Ask an administrator to assign you access, then try logging in again.
          </p>
        </div>
      </div>
    </div>
  </div>
</main>
`;
